.github/actions/checkout-repository/action.yaml

```diff
@@ -32,7 +32,7 @@ runs:
     - name: Checkout Repository
       uses: actions/checkout@v4
       with:
-        repository: "NHSDigital/${{ inputs.repository }}"
+        repository: "NHSDigital/${{ inputs.repository == '' && github.event.repository.name || inputs.repository }}"
```

Comment:
Prefacing all comments with the caveat that this is day two for me after a long lay off ...
Do we need this logic as the default if repository is not set is github.event.repository.name ?

Reply:
I believe there are some peculiarities in github actions where an input being an empty string is not the same as an input being absent, hence this extra check.